ruby/rspec/tutorial/let-vs-let.md

```diff
@@ -0,0 +1,34 @@
+# let vs let!
+
+Understanding the difference between `let` and `let!` is fundamental when starting to learn RSpec.
+
+Use let to define a memoized helper method. The value will be cached
+across multiple calls in the same example but not across examples.
+
+Note that let is lazy-evaluated: it is not evaluated until the first time
+the method it defines is invoked. You can use let! to force the method's
+invocation before each example.
+
+### When should `let!` be used?
+- When there's a need to trigger an insertion into the database before running an example block.
+- An alternative to `let!` is invoking the variable from a `before` or `subject` block, or even from inside an example. This practice is discouraged.
+
+### Example
+
+#### Bad
+```
+let!(:user) { create(:user) }
+
+let!(:membership) { create(:membership, user: user) }
```

Comment:
maybe you could add an example of a non active record creation, like `let!(:params) { { 'some_params': 'first_param' } }` or something like that, to show other cases of wrong and good examples

Reply:
yeah, let's try not to associate `let!` with ActiveRecord usages only